tests/functionaltests/FunctionalTestHelpers.mm

```diff
@@ -14,14 +14,36 @@
 //
 //******************************************************************************
 
-#import "FunctionalTestHelpers.h"
+#import <Logger.h>
+#import <FunctionalTestHelpers.h>
+#import <AppDelegate.h>
+
 #include "UIKit/UIApplication.h"
+#include "UIViewInternal.h"
 #import <Starboard/SmartTypes.h>
+#import <StringHelpers.h>
 #import <UWP/WindowsApplicationModel.h>
 
-// Prevents UIApplication state from carrying over between functional tests
-void FunctionalTestCleanupUIApplication() {
-    [[UIApplication sharedApplication] _destroy];
+// This is a method that UIKit exposes for the test frameworks to use.
+extern "C" void UIApplicationInitialize(const wchar_t*, const wchar_t*);
+
+// Launches the functional test app
+bool FunctionalTestSetupUIApplication() {
+    RunSynchronouslyOnMainThread(^{
+        // The name of our default 'AppDelegate' class
+        UIApplicationInitialize(nullptr, Strings::NarrowToWide<std::wstring>(NSStringFromClass([AppDelegate class])).c_str());
```

Comment:
nice catch! maybe we don't need the extra function after all.

---
In reply to: [96552727](https://github.com/Microsoft/WinObjC/pull/1716#discussion_r96552727) [](ancestors = 96552727)